v2/flat_map.go

```diff
@@ -0,0 +1,10 @@
+package pie
+
+// FlatMap maps each element into a new slice, then flattens the result.
+func FlatMap[T any, U any](ss []T, fn func(T) []U) []U {
+	if ss == nil {
+		return nil
+	}
+
+	return Flat(Map(ss, fn))
```

Comment:
Can't this be achieved by chaining operations? Why does this need a dedicate function?